src/librustc_middle/ty/steal.rs

```diff
@@ -41,4 +41,8 @@ impl<T> Steal<T> {
         let value = value_ref.take();
         value.expect("attempt to read from stolen value")
     }
+
+    pub fn is_stolen(&self) -> bool {
+        self.value.borrow().is_none()
+    }
```

Comment:
doesn't this make the query system unsound? :sweat_smile:

The call graph and possibly the result of a query now changes depending on which queries where called before.

Reply:
Uh, possibly, yeah :)

I think here the impact is limited to this lint though, so maybe it's not too bad? Otherwise we could maybe run the lint after `optimized_mir`?

Reply:
hmm, the "most correct" solution is probably to collect all `inevitable_calls_in_body` using a query run after `mir_built` and run this lint in `mir_validated` or sth :thinking: